paasta_tools/setup_prometheus_adapter_config.py

```diff
@@ -0,0 +1,249 @@
+    resources: Dict[str, str]
+    # the actual query we want to send to Prometheus to use for scaling
+    metricsQuery: str
+
+
+class PrometheusAdapterConfig(TypedDict):
+    """
+    Typed version of the Prometheus adapter configuration dictionary.
+    """
+
+    rules: List[PrometheusAdapterRule]
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description="Syncs the Prometheus metric adapter config with soaconfigs.",
+    )
+
+    parser.add_argument(
+        "-d",
+        "--soa-dir",
+        dest="soa_dir",
+        metavar="SOA_DIR",
+        default=Path(DEFAULT_SOA_DIR),
+        help="Directory to read service configs from. Default is %(default)s.",
+        type=Path,
+    )
+    # TODO: do we need to be able to pass multiple clusters in?
+    parser.add_argument(
+        "-c",
+        "--cluster",
+        dest="cluster",
+        help="PaaSTA cluster to generate configs for.",
+        required=True,
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="store_true",
+        dest="verbose",
+        default=False,
+        help="Enable verbose logging.",
+    )
+    parser.add_argument(
+        "--dry-run",
+        dest="dry_run",
+        action="store_true",
+        default=False,
+        help="Enable verbose logging.",
+    )
+
+    return parser.parse_args()
+
+
+def should_create_uwsgi_scaling_rule(
+    instance: str, instance_config: Dict[str, Any],
+) -> Tuple[bool, Optional[str]]:
+    """
+    Determines whether we should configure the prometheus adapter for a given service.
+    Returns a 2-tuple of (should_create, reason_to_skip)
+    """
+    return False, ""
+
+
+def create_instance_uwsgi_scaling_rule(
+    service: str, instance: str, instance_config: Dict[str, Any], paasta_cluster: str
+) -> PrometheusAdapterRule:
+    """
+    Creates a Prometheus adapter rule config for a given service instance.
+    """
+    return {
+        "name": {"as": ""},
+        "seriesQuery": "",
+        "resources": {"template": "kube_<<.Resource>>"},
+        "metricsQuery": "",
+    }
+
+
+def create_prometheus_adapter_config(
+    paasta_cluster: str, soa_dir: Path
+) -> PrometheusAdapterConfig:
+    """
+    Given a paasta cluster and a soaconfigs directory, create the necessary Prometheus adapter
+    config to autoscale services.
+    Currently supports the following metrics providers:
+        * uwsgi
+    """
+    rules: List[PrometheusAdapterRule] = []
+    # we don't know ahead of time what services are autoscaled, so we need to figure out
+    # what services are running in our cluster and then check if they're autoscaled
+    for service_config_path in soa_dir.glob(f"*/kubernetes-{paasta_cluster}.yaml"):
+        service_name = str(service_config_path.relative_to(soa_dir).parent)
+        service_config = yaml.safe_load(service_config_path.read_text())
+        for instance, instance_config in service_config.items():
```

Comment:
I think you may want to use `utils.get_services_for_cluster` and PaastaServiceConfigLoader. This will do more processing than just loading the yaml (it'll create KubernetesDeploymentConfigs), but that's probably a good thing. It should do the `_` prefix filtering for you, and then you won't be reimplementing the interpretation of the autoscaling dict.

Reply:
https://github.com/Yelp/paasta/pull/3026/commits/66caa8f747f8186b131849f13360a88898e7004b does this now